src/main/java/org/truffleruby/core/string/FrozenStrings.java

```diff
@@ -24,10 +24,10 @@
 
     private final RubyContext context;
 
-    private final Map<RopeHolder, DynamicObject> frozenStrings = new WeakHashMap<>();
+    private final Map<Rope, DynamicObject> frozenStrings = new WeakHashMap<>();
     private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
-    private final Set<RopeHolder> keys = new HashSet<>();
+    private final Set<Rope> keys = new HashSet<>();
```

Comment:
Does this commit enables to remove this `keys` set?
Maybe for another PR?

Reply:
See my comment below. I will tackle this in another PR when this is merged.

Reply:
I think removing the `keys` is correct, but I wasn't entirely sure and was going to let @chrisseaton handle that given he just did some major work to `FrozenStrings`.